### DISCUSSION TITLE
How to split stores into several files?

### DISCUSSION THREAD

**Discussion by hugoattal** [vuejs/pinia#709] [Help and Questions] [answered]
My store is getting pretty big, and I'd like to separate it into different files.

Thing is, I don't want to lose the type inferring (it's soooo useful)

I totally can do something like this:
```ts
// store.ts
import { doThing } from "myActions"

export const useMyStore = defineStore({
    id: "my-store",
    actions: {
        doThing(data: any) {
            doThing(this, any);
        }
    }
    state: () => ({
        myStuff: any
    })
});

export type TMyStore = ReturnType<typeof useMyStore >


// myActions.ts
import { TMyStore  } from "store";

export function doThing(store: TMyStore, data: any): void {
    store.myStuff = data;
}
```

But I would prefer my store to look like this

```ts
// store.ts
import { doThing } from "myActions"

export const useMyStore = defineStore({
    id: "my-store",
    actions: {
        doThing
    }
    state: () => ({
        myStuff: any
    })
});


// myActions.ts
// ???
```

Is this something possible?

**Comment by posva** [comment, marked as answer]
I would avoid splitting the store into multiple files. If anything split the store into multiple stores or extract heavy functions outside of actions and rely on passing the necessary information of the store as arguments (instead of passing the whole store)

**Reply by hugoattal** [reply]
Thank you very much, I'll split my store then!
